dynamic_instrumentation/src/instrument_memory.rs

```diff
@@ -236,16 +236,64 @@ impl<'a, 'tcx: 'a> CollectFunctionInstrumentationPoints<'a, 'tcx> {
         find_instrumentation_def(self.tcx, self.runtime_crate_did, name)
     }
 
-    fn should_instrument(&self, local: Local, context: PlaceContext) -> bool {
-        let is_unsafe_ptr = self.body.local_decls[local].ty.is_unsafe_ptr();
-        is_unsafe_ptr && context.is_use()
+    fn is_shared_ptr(&self, ty: &TyS<'tcx>) -> bool {
+        ty.is_region_ptr() && !ty.is_mutable_ptr()
+    }
```

Comment:
It doesn't need `&self`, and then it could be called with `Self::is_shared_ptr`.